### DISCUSSION TITLE
local lyrics support  and netease cloud music.

### DISCUSSION THREAD

**Discussion by funnyleolee** [tramhao/termusic#391] [General]
Is support local lyrics.  And how to search NetEase Cloud Music.

**Comment by tramhao** [comment]
I remember that only album photo can be supported to load from local file. Lyrics need to embed into file. You can use tools to embed it. In order to search NetEase, select a file, enter tag editor by press `t`. In there you can search for artist or song name.

**Comment by hasezoey** [comment]
Currently termusic only loads LRC content if it is embedded into the file as Text(`UnsynchronizedTextFrame`) in the music file.

It *would* be possible to add loading a `.lrc` file in the future, but with how termusic currently handles it, it would load it and then embed it (and doing other metadata changes). (i found this out by adding such a way for quick testing of lrc parsing)

<details>
<summary>Example of my debug adding lrc loading</summary>

```diff
diff --git a/lib/src/track.rs b/lib/src/track.rs
index d87ce755..2551cd1a 100644
--- a/lib/src/track.rs
+++ b/lib/src/track.rs
@@ -161,7 +161,7 @@ impl Track {
 
         let parent_folder = get_parent_folder(path);
 
-        if let Ok(files) = std::fs::read_dir(parent_folder) {
+        if let Ok(files) = std::fs::read_dir(&parent_folder) {
             for f in files.flatten() {
                 let path = f.path();
                 if let Some(extension) = path.extension() {
@@ -172,6 +172,22 @@ impl Track {
             }
         }
 
+        // try to look for a lrc file with the same name, only different extension
+        if song.parsed_lyric().is_none() {
+            if let Some(file_stem) = path.file_stem() {
+                let lrc_path = {
+                    let mut filename = file_stem.to_os_string();
+                    filename.push(".lrc");
+                    parent_folder.join(filename)
+                };
+
+                if lrc_path.exists() {
+                    let lyric_str = std::fs::read_to_string(lrc_path)?;
+                    song.parsed_lyric = Lyric::from_str(&lyric_str).ok();
+                }
+            }
+        }
+
         Ok(song)
     }

```

</details>
